src/server/components/dashboard/controllers/listsExportController.ts

```diff
@@ -0,0 +1,94 @@
+import type { NextFunction, Request, Response } from "express";
+import type { ListItem } from "server/models/types";
+import type { ListItemJsonData } from "server/models/listItem/providers/deserialisers/types";
+
+import { prisma } from "server/models/db/prisma-client";
+import { json2csv } from "json-2-csv";
+import { Readable } from "node:stream";
+import { startCase } from "lodash";
+import { format } from "date-fns";
+import { logger } from "server/services/logger";
+import { AuditEvent } from "@prisma/client";
+
+type ListWithJsonData = Array<ListItem & { jsonData: ListItemJsonData }>;
+
+export async function listsExportController(req: Request, res: Response, next: NextFunction) {
+  try {
+    const [result] = await prisma.$transaction([
+      prisma.list.findUnique({
+        where: {
+          id: Number(req.params.listId),
+        },
+        include: {
+          items: true,
+          country: true,
+        },
+      }),
+      prisma.audit.create({
+        data: {
+          type: "list",
+          auditEvent: AuditEvent.LIST_EXPORTED,
+          jsonData: {
+            listId: Number(req.params.listId),
+            userId: req.user?.id,
+            userEmail: req.user?.emailAddress,
+          },
+        },
+      }),
+    ]);
+
+    const formattedItems = formatProviderData((result?.items as ListWithJsonData) ?? []);
```

Comment:
You can also opt for a more fp style. It makes it nicer to test imo (I guess tests are not really necessary here it's just doing some combination of spread/remapping keys. But if you wanted to test, you could test key order or what happens when address is missing etc)

```
const formattedItems = result.items.map(formatForCSV)


function formatForCSV(item: ListItem) {
// do any casting here if you need to so it keeps the above call cleaner.
const { type, jsonData, ...rest } = item;
    const {
      organisationName,
      "address.firstLine": addressFirstLine,
      "address.secondLine": addressSecondLine,
      ...otherFields
    } = jsonData as ListItemJsonData;
    return {
      organisationName,
      ...otherFields,
      addressFirstLine,
      addressSecondLine,
      ...rest,
    };
  }

}
```